### PR TITLE
Refactor to rename viewer to pilot

### LINKED ISSUE #400: Change the "viewer" to use an official name
The "viewer" GUI app was named like that because "viewer" is a placeholder name.  After fixing the Mac app icon (PR #398), we can start to give the GUI app an official name.

It could take a while.  This issue is created to track the progress of naming.

---

Update on 20th December 2024: I decide to use "pilot" as the new name.  Start to change the code.  (Ref: discussion #443)

### PR DESCRIPTION
**Pull request by yungyuc**
Rename the Python module `modmesh.view` to package `modmesh.pilot`.  `modmesh.pilot` will host all code related to GUI and visualization.

Also rename the GUI executable from `viewer` to `pilot`.

### CODE REVIEW

File: modmesh/pilot/_gui.py
Comment (yungyuc) [inline modmesh/pilot/_gui.py:1]: This file `modmesh/pilot/_gui.py` comes from `modmesh/view.py`.

File: modmesh/pilot/__init__.py
Comment (yungyuc) [inline modmesh/pilot/__init__.py:1]: `modmesh.pilot` will be a drop-in replacement to `modmesh.view`.

**Review by yungyuc — COMMENTED** [review]
The PR only changes Python module name and GUI executable name.  Other source file names are not changed.

Change of file names will be included in a followup PR.

**Comment by yungyuc** [comment]
Merged the PR.  Start to implement for issue #400.

**Comment by yungyuc** [comment]
@j8xixo12 Please help take a look when you have time.  We can fix issues with followup PRs.

**Comment by j8xixo12** [comment]
> @j8xixo12 Please help take a look when you have time. We can fix issues with followup PRs.

No problem, I will take a look on this PR and #447.

**Pull request merged**